### PR TITLE
RSS/Atom support for Orgs

### LINKED ISSUE #569: Basic RSS support
Add an RSS feed for
* [x] Repo (#19055)
* [x] User (#16002)
* [x] Organisation (#17714)

### PR DESCRIPTION
**Pull request by 6543**
close partialy #569

* [x] Backend
* [x] Tests
* [x] Org Rss
* [ ] ~~Repo RSS~~

### CODE REVIEW

File: models/action.go [hunk 1]
```diff
@@ -315,29 +316,40 @@ func (a *Action) GetIssueContent() string {
...
 	}
 
 	cond, err := activityQueryCondition(opts)
 	if err != nil {
 		return nil, err
 	}
 
-	actions := make([]*Action, 0, setting.UI.FeedPagingNum)
+	sess := db.GetEngine(db.DefaultContext).Where(cond)
+	if opts.Page >= 0 {
```
Comment (wxiaoguang) [inline models/action.go]: `opts.Page <= 0` ?

Comment (6543) [inline models/action.go]:
```suggestion
	if opts.Page <= 0 {
```

Comment (6543) [reply models/action.go]: ugh .. good catch

File: routers/web/feed/profile.go [hunk 2]
```diff
@@ -23,31 +23,34 @@ func RetrieveFeeds(ctx *context.Context, options models.GetFeedsOptions) []*mode
...
 
-	userCache := map[int64]*user_model.User{options.RequestedUser.ID: options.RequestedUser}
-	if ctx.User != nil {
-		userCache[ctx.User.ID] = ctx.User
-	}
-	for _, act := range actions {
-		if act.ActUser != nil {
-			userCache[act.ActUserID] = act.ActUser
+	// TODO: move load repoOwner of act.Repo into models.GetFeeds->loadAttributes()
+	{
```
Comment (zeripath) [inline routers/web/feed/profile.go:27]: Why this?

Comment (6543) [reply routers/web/feed/profile.go:27]: note to refactor - but that would make this pull to large, next pull will follow for sure ... to round up and make repo rss posible (backend should already support it but its not exported & not tested)

File: models/action.go (hunk 1 above)
Comment (wxiaoguang) [reply models/action.go]: Actually, we do not need these `Page <= 0` checks again and again.

The framework already does:

image

File: routers/web/feed/profile.go (hunk 2 above)
Comment (6543) [reply routers/web/feed/profile.go:27]: @zeripath should I add the refactor here?

Comment (zeripath) [reply routers/web/feed/profile.go:27]: I was meaning the placing of this section of code within a {} block

Comment (6543) [reply routers/web/feed/profile.go:27]: Thats the block that should be moved ... it does not have any efect on how it works

**Review by zeripath — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

File: routers/web/feed/profile.go (hunk 2 above)
Comment (6543) [reply routers/web/feed/profile.go:27]: -> #19055

or to be specific: https://github.com/go-gitea/gitea/pull/19055/commits/ed80cfc5cc38e7d14e1000e00e7c4047fb851013

**Pull request merged**